geniza/corpus/models.py

```diff
@@ -1293,8 +1293,35 @@ def merge_with(self, merge_docs, rationale, user=None):
         needs_review = [self.needs_review] if self.needs_review else []
 
         for doc in merge_docs:
+            # handle document dates validation before making any changes;
+            # mismatch should result in exception (caught by DocumentMerge.form_valid)
+            if (
+                doc.doc_date_standard
+                and self.doc_date_standard
+                and self.doc_date_standard != doc.doc_date_standard
+            ) or (
+                doc.doc_date_original
+                and self.doc_date_original
+                and self.doc_date_original != doc.doc_date_original
+            ):
+                raise ValidationError(
+                    "Merged documents must not contain conflicting dates; resolve before merge"
+                )
+
             # add any tags from merge document tags to primary doc
             self.tags.add(*doc.tags.names())
+
+            # if not in conflict, migrate dates to result document
```

Comment:
this is either copying over dates where none exist or copying over exact duplicates? that makes sense, might be worth slightly expanding the comment